src/lib/storage/segment_accessor.hpp

```diff
@@ -85,12 +87,20 @@ class SingleChunkReferenceSegmentAccessor : public BaseSegmentAccessor<T> {
   explicit SingleChunkReferenceSegmentAccessor(const ReferenceSegment& segment)
       : _segment{segment},
         _chunk_id((*_segment.pos_list())[ChunkOffset{0}].chunk_id),
-        _accessor{create_segment_accessor<T>(
-            segment.referenced_table()->get_chunk(_chunk_id)->get_segment(_segment.referenced_column_id()))} {}
+        // If *_segment.pos_list())[ChunkOffset{0}] is NULL, its chunk_id is INVALID_CHUNK_OFFSET. When the
+        // SingleChunkReferenceSegmentAccessor is used, all entries reference the same chunk_id (INVALID_CHUNK_OFFSET).
+        // Therefore, we can safely assume that all other entries are also NULL and always return std::nullopt.
+        _accessor((*_segment.pos_list())[ChunkOffset{0}].is_null()
+                      ? nullptr
+                      : create_segment_accessor<T>(segment.referenced_table()->get_chunk(_chunk_id)->get_segment(
+                            _segment.referenced_column_id()))) {}
 
   const std::optional<T> access(ChunkOffset offset) const final {
-    const auto referenced_chunk_offset = (*_segment.pos_list())[offset].chunk_offset;
+    if (!_accessor) {
```

Comment:
Did you benchmark this? Adding a branch in a hot method seems like it might influence the performance. Maybe add a class-internal `NullAccessor` that always returns `std::nullopt`? That way we would simply direct the virtual method call somewhere else.